torch/utils/data/sampler.py

```diff
@@ -48,24 +52,32 @@ def __iter__(self):
     #     (@ssnl verifies that this works on at least Python 3.7.)
 
 
-class SequentialSampler(Sampler):
+class SizedSampler(Sampler[T_co], Sized):
```

Comment:
According to the note at https://github.com/pytorch/pytorch/blob/2b29feace4d4711190ad0a5012ba517f45e0e336/torch/utils/data/sampler.py#L23-L48 , we should not be providing a default implementation of `__len__`. Do we need SizedSampler for all the types to check out? Scrolling through this PR it looks like SizedSampler is only used to construct the other Samplers